### PR TITLE
feat: support torch.ops.aten.sum.(default and dim_IntList) dynamo converter

### LINKED ISSUE #2244: Implement converter for `torch.ops.aten.sum.dim_IntList`
Schema:
https://github.com/pytorch/pytorch/blob/bdecdfd202df3fa25fd9998070fd19fee4b14971/aten/src/ATen/native/native_functions.yaml#L5478-L5487

### PR DESCRIPTION
**Pull request by zewenli98**
# Description

Support `torch.ops.aten.sum.dim_IntList` dynamo converter

Fixes #2244 

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/impl/reduce.py [hunk 1]
```diff
@@ -33,3 +33,29 @@ def amax(
     )
     set_layer_name(layer, target, name, source_ir)
     return layer.get_output(0)
+
+
+def sum_dim_IntList(
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/conversion/impl/reduce.py:38]: this seems over specialized, what are the other variants of sum? Can we combine the implementations?

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/impl/reduce.py]: @narendasan Thanks for your suggestion! I updated to support both `aten.sum.default` and `aten.sum.dim_IntList`.

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/impl/reduce.py]: @narendasan lmk if more changes are needed! thanks!

**Review by gs-olive — APPROVED** [review]
Looks good to me!

**Pull request merged**